apps/backend/src/lib/rds.ts

```diff
@@ -103,31 +167,20 @@ export class RDS {
         return scheduleId;
     }
 
-    /**
-     * If the guest user with the username in the userData object doesn't already exist,
-     * create a new guest user and populate their data. Otherwise, returns null.
-     */
-    static async insertGuestUserData(db: DatabaseOrTransaction, userData: User): Promise<string | null> {
-        return db.transaction(async (tx) => {
-            const userId = await RDS.guestUserIdWithNameOrNull(tx, userData.id);
-            if (userId) return null;
-            return RDS.upsertGuestUserData(tx, userData);
-        });
-    }
-
     /**
      * Does the same thing as `insertGuestUserData`, but also updates the user's schedules and courses if they exist.
      *
      * @param db The Drizzle client or transaction object
      * @param userData The object of data containing the user's schedules and courses
      * @returns The user's ID
      */
-    static async upsertGuestUserData(db: DatabaseOrTransaction, userData: User): Promise<string> {
+    static async upsertUserData(db: DatabaseOrTransaction, userData: User): Promise<string> {
         return db.transaction(async (tx) => {
-            const userId = await this.createGuestUserOptional(tx, userData.id);
+            let userId = userData.id;
 
-            if (userId === undefined) {
-                throw new Error(`Failed to create guest user for ${userData.id}`);
+            let user = await this.getUserById(db, userId);
```

Comment:
Don't you get errors in the editor for this?
```suggestion
            const user = await this.getUserById(db, userId);
```